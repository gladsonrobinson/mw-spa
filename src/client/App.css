.main-container {
  text-align: center;
}

.main-container .header {
  background-color: #282c34;
  min-height: 10px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.main-container .header a {
  color: #ffffff;
  margin: 0 20px;
}

.container {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}
@media (min-width: 768px) {
  .container {
    max-width: 738px;
  }
}
@media (min-width: 992px) {
  .container {
    max-width: 940px;
  }
}
@media (min-width: 1200px) {
  .container {
    max-width: 1170px;
  }
}

.container.mw-map {
  display: flex;
  flex-direction: row;
  margin-top: 10px;
}
