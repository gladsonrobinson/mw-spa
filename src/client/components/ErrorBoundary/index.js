import React from "react";

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasComponentError: false };
  }

  componentDidCatch(error, info) {
    this.setState({ hasComponentError: true });
  }

  render() {
    if (this.state.hasComponentError) {
      return <h4 className="component--error">Some thing went wrong</h4>
    }
    return this.props.children;
  }
}
