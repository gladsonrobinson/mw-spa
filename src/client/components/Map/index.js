import React from "react";
import { compose, withProps } from "recompose"
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"

const Map = compose(
  withProps({
    googleMapURL: "https://maps.googleapis.com/maps/api/js?key=AIzaSyBY448uB6sUyz3ETiIYwciXQdpAOoWGeGQ&v=3.exp&libraries=geometry,drawing,places",
    loadingElement: <div style={{ height: `100%` }} />,
    containerElement: <div style={{ height: `400px`, flex: 1 }} />,
    mapElement: <div style={{ height: `100%` }} />,
  }),
  withScriptjs,
  withGoogleMap
)((props) =>
  <GoogleMap
    defaultZoom={5}
    defaultCenter={{ lat: 51.1657, lng: 10.4515 }}
  >
    {
      props.mapMarkers.length && 
      props.mapMarkers.map(item => 
        <Marker position={{ lat: item.latLng.lat, lng: item.latLng.lng }} key={item._id} /> )
    }
  </GoogleMap>
);

export default Map;
