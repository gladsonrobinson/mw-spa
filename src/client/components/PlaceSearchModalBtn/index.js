import React from "react";
import ReactModal from "react-modal";
import LocationSearchInput from "../LocationSearchInput";
import "./PlaceSearchModalBtn.css";

export default class PlaceSearchModalBtn extends React.Component {
  render() {
    return (
      <div className="add-map">
        <button
          className="add-map__button"
          onClick={this.props.handleOpenModal}
        >
          ADD MAP
        </button>

        <ReactModal isOpen={this.props.showModal} contentLabel="Place Search">
          <div>
            <button onClick={this.props.handleCloseModal}>Close Modal</button>

            <LocationSearchInput onSelect={this.props.onPlaceSelect} />
          </div>
        </ReactModal>
      </div>
    );
  }
}
