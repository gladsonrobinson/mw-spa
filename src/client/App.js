import React, { Component } from "react";
import "./App.css";
import { Provider } from "react-redux";
import store from "./redux/store";
import AppCtnr from "./containers/App";

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <AppCtnr />
      </Provider>
    );
  }
}
