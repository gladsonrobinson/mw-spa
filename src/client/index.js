import React from "react";
import ReactDOM from "react-dom";
import "./index.css";
import App from "./App";
console.log("map key", process.env.REACT_APP_MAP_KEY)

ReactDOM.render(<App />, document.getElementById("root"));
