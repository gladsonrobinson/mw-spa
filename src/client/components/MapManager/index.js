import React from "react";
import PlaceSearchModalBtn from "../PlaceSearchModalBtn";
import "./map-manager.css";

export default class MapManager extends React.PureComponent {
  onDelectClick = id => {
    this.props.removeMapMarker(id);
  };
  render() {
    return (
      <div className="map-manager">
        <div>
          <PlaceSearchModalBtn
            showModal={this.props.showModal}
            handleOpenModal={this.props.showPlaceSearchModal}
            handleCloseModal={this.props.hidePlaceSearchModal}
            onPlaceSelect={this.props.onPlaceSelect}
          />
        </div>
        <div className="map-manager-markers">
          {this.props.mapMarkers.length > 0 &&
            this.props.mapMarkers.map(item => {
              return (
                <div className="map-manager-markers__item" key={item._id}>
                  <h4 className="map-manager-markers__item__address">
                    {item.address}
                  </h4>
                  <div className="map-manager-markers__item__latlng">
                    Latitude: {item.latLng.lat}
                  </div>
                  <div className="map-manager-markers__item__latlng">
                    Longitude: {item.latLng.lng}
                  </div>
                  <button
                    className="map-manager-markers__item__delete"
                    onClick={this.onDelectClick.bind(null, item._id)}
                  >
                    Delete
                  </button>
                </div>
              );
            })}
        </div>
      </div>
    );
  }
}
