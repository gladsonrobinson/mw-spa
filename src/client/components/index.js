import Wrapper from "./Wrapper";
import SearchBox from "./SearchBox";
import Map from "./Map";
import MapManager from "./MapManager";

export { Wrapper, SearchBox, Map, MapManager };
