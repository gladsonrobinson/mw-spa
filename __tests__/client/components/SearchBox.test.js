import React from "react";
import { SearchBox } from "../../../src/client/components";
import { shallow } from "enzyme";
import * as sinon from "sinon";

import renderer from "react-test-renderer";

describe("Search Component", () => {
  let clock;

  beforeEach(() => {
    clock = sinon.useFakeTimers();
  });

  afterEach(() => {
    clock.restore();
  });

  it("Should render the component", () => {
    const searchFunction = jest.fn();
    const wrapper = shallow(<SearchBox onSearch={searchFunction} />);
    expect(wrapper.exists()).toBe(true);
  });

  it("should render correctly the component", () => {
    const searchFunction = jest.fn();
    const SearchBoxComponent = renderer
      .create(<SearchBox onSearch={searchFunction} />)
      .toJSON();
    expect(SearchBoxComponent).toMatchSnapshot();
  });

  it("Should not trigger search function on input change before 300 m sec", () => {
    const searchFunction = jest.fn();
    const wrapper = shallow(<SearchBox onSearch={searchFunction} />);
    wrapper.find("input").simulate("change", {
      target: {
        value: "hello"
      },
      persist: () => {}
    });

    expect(searchFunction).toHaveBeenCalledTimes(0);
  });

  it("Should trigger search function on input field change after debounce of 300 m sec", () => {
    const searchFunction = jest.fn();
    const wrapper = shallow(<SearchBox onSearch={searchFunction} />);
    wrapper.find("input").simulate("change", {
      target: {
        value: "hello"
      },
      persist: () => {}
    });

    clock.tick(300);
    expect(searchFunction).toHaveBeenCalledTimes(1);
  });
});
