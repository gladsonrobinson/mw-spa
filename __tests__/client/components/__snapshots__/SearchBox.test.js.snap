// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render correctly the component 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Search Movie Name."
    type="text"
  />
</div>
`;
